spring-vault-core/src/main/java/org/springframework/vault/core/VaultTransitOperations.java

```diff
@@ -123,6 +124,32 @@
 	String encrypt(String keyName, byte[] plaintext, VaultTransitContext transitRequest);
 
 	/**
+	 * Encrypts the provided list of plaintext using the named key.
+	 * The encryption is done using transit backend's batch operation.
+	 *
+	 * works with Vault 0.6.5 and later.
+	 *
+	 * @param keyName must not be empty or {@literal null}.
+	 * @param plaintexts a list of non-empty (or {@literal null}) plaintext.
+	 * @return list of cipher text in the same order as in plaintexts.
+	 */
+	List<Map<String, String>> encrypt(String keyName, List<String> plaintexts);
```

Comment:
Returning a `Map` of `String`s isn't quite expressive. It's not obvious what the key and what the values are. I'd rather return a `List<String>`. You give the method an ordered list of plaintexts and receive an ordered list of ciphertexts.

Reply:
makes lot of sense. will work on it.